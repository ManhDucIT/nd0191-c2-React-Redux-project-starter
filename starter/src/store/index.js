import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../slices/authSlice";
import usersReducer from "../slices/usersSlice";
import questionsReducer from "../slices/questionsSlice";

const store = configureStore({
    reducer: {
        auth: authReducer,
        users: usersReducer,
        questions: questionsReducer
    }
});

export default store;