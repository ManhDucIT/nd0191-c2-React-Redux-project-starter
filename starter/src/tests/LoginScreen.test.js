import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import { render, screen, fireEvent } from '@testing-library/react';
import store from "../store/index";
import Login from '../screens/LoginScreen';

describe("LoginScreen", () => {
    it("Should render a snapshot for Login screen", async () => {
        const component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Login />
                </Provider>
            </MemoryRouter>
        );

        expect(component).toMatchSnapshot();
    });

    it("Will display an error if username is empty", async () => {
        const component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Login />
                </Provider>
            </MemoryRouter>
        );
        const passwordInput = screen.getByTestId('password-field');
        fireEvent.change(passwordInput, { target: { value: 'abc123' } });
        const loginButton = screen.getByTestId('submit-button');
        fireEvent.click(loginButton);
        expect(component.queryByTestId("error-username")).toBeInTheDocument();
        expect(loginButton).toBeInTheDocument();
    });

    it('Should render user name field, password field, and submit button', () => {
        render(
            <MemoryRouter>
                <Provider store={store}>
                    <Login />
                </Provider>
            </MemoryRouter>
        );
      
        const usernameElement = screen.getByTestId('username-field');
        expect(usernameElement).toBeInTheDocument();
      
        const passwordElement = screen.getByTestId('password-field');
        expect(passwordElement).toBeInTheDocument();
    
        const submitElement = screen.getByTestId('submit-button');
        expect(submitElement).toBeInTheDocument();
    });
});