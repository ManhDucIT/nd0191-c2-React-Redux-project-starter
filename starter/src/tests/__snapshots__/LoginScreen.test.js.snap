// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Should render a snapshot for Login screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="login-box"
        >
          <h1>
            Employee Polls
          </h1>
          <img
            alt="Employee Polls Logo"
            src="employee_polls_logo.svg"
          />
          <h2>
            Login
          </h2>
          <form>
            <div>
              <input
                class=""
                data-testid="username-field"
                placeholder="Username"
                type="text"
                value="sarahedo"
              />
            </div>
            <div>
              <input
                class=""
                data-testid="password-field"
                placeholder="Password"
                type="password"
                value="password123"
              />
            </div>
            <button
              data-testid="submit-button"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="login-box"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          alt="Employee Polls Logo"
          src="employee_polls_logo.svg"
        />
        <h2>
          Login
        </h2>
        <form>
          <div>
            <input
              class=""
              data-testid="username-field"
              placeholder="Username"
              type="text"
              value="sarahedo"
            />
          </div>
          <div>
            <input
              class=""
              data-testid="password-field"
              placeholder="Password"
              type="password"
              value="password123"
            />
          </div>
          <button
            data-testid="submit-button"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
